server/src/main/java/org/opensearch/common/logging/OpenSearchJsonLayout.java

```diff
@@ -91,26 +94,34 @@ public class OpenSearchJsonLayout extends AbstractStringLayout {
 
     private final PatternLayout patternLayout;
 
-    protected OpenSearchJsonLayout(String typeName, Charset charset, String[] opensearchMessageFields) {
+    protected OpenSearchJsonLayout(String typeName, Charset charset, String[] opensearchMessageFields, int maxMessageLength) {
         super(charset);
         this.patternLayout = PatternLayout.newBuilder()
-            .withPattern(pattern(typeName, opensearchMessageFields))
+            .withPattern(pattern(typeName, opensearchMessageFields, maxMessageLength))
             .withAlwaysWriteExceptions(false)
             .build();
     }
 
-    private String pattern(String type, String[] opensearchMessageFields) {
+    private String pattern(String type, String[] opensearchMessageFields, int maxMessageLength) {
         if (Strings.isEmpty(type)) {
             throw new IllegalArgumentException("layout parameter 'type_name' cannot be empty");
         }
+
+        String messageFormat = "%m";
+        if (maxMessageLength < 0) {
```

Comment:
👍 , to you think the upper limit would also make sense to let say 500k? (going with 2g messages is too much I think)